nav2_amcl/CMakeLists.txt

```diff
@@ -1,6 +1,14 @@
 cmake_minimum_required(VERSION 3.5)
 project(nav2_amcl)
 
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
Revert to c++14. ROS2 doesnt support c++17 yet and should only be used when specifically required until that migration is made.